project.clj

```diff
@@ -7,9 +7,21 @@
                  [instaparse/instaparse "1.4.10"]]
   :profiles {:dev {:dependencies [[criterium/criterium "0.4.5"] ;; benchmark
                                   [org.clojure/test.check "0.10.0"]]
-                   :plugins      [[jonase/eastwood "0.3.5"]]}}
+                   :plugins      [[jonase/eastwood "0.3.5"]]}
+             :provided {:dependencies [[org.clojure/clojurescript "1.10.520"]]}}
   :test-selectors {:default     (fn [m] (not (some #{:benchmark} (keys m))))
                    :benchmark   :benchmark}
+  :cljsbuild {:builds
+              {"dev" {:source-paths ["src" "test"]
+                      :compiler {:main parcera.test-runner
```

Comment:
out of ignorance .... how do you run this ? I would expect something similar to `lein test` but I guess that would run the tests on the JVM 🤔 
I guess we should update the `.travis.yml` to also test the cljs code ?

Reply:
I run the tests using Calva, which in turns uses its ClojureScript REPL, so I jack it in with **Leiningen + Nashorn**.

I also run it manually using: ```sh
$ lein cljsbuild once
$ node target/out/tests.js
```

And also `lein test` runs the ClojureScript tests too. (Not sure how though.)

Reply:
> And also `lein test` runs the ClojureScript tests too. (Not sure how though.)

That doesnt seem to be the case ... I see in the Travis CI logs for `compiling clojurescript` but not for testing it 🤔

Reply:
I would guess it needs node. Maybe if that isn't present, that it skips the CLJS tests. I would expect a message telling me that, though, so maybe something else is going on.

When I run `lein test` on my machine it first runs the tests using Clojure, then using ClojureScript. As in https://github.com/carocad/parcera/pull/3#issuecomment-538800429

Reply:
I was just reading this [website](https://github.com/emezeske/lein-cljsbuild/blob/1.1.7/doc/TESTING.md) regarding test on cljs. From what I can see the current setup relies on `hooks` which are deprecated and we dont really know why it doesnt work. Given that I will propose that you explicitedly call the cljs test suite like `lein cljsbuild test` on the `.travis.yml` file. Let me know what you think about it :)

Reply:
Makes total sense. I removed the hook and renamed the `:test-commands` entry to "test”, because, well, I don't know why I had named it ”dev”.

I also then could remove the global `:cljsbuild` entry, since it is only needed in dev.

Please note that you still need nodejs on the machine running the tests.